test/object-store/util/test_file.cpp

```diff
@@ -374,6 +377,43 @@ TestAppSession::~TestAppSession()
     }
 }
 
+std::vector<bson::BsonDocument> TestAppSession::get_documents(SyncUser& user, const std::string& object_type,
+                                                              size_t expected_count) const
+{
+    app::MongoClient remote_client = user.mongo_client("BackingDB");
+    app::MongoDatabase db = remote_client.db(m_app_session->config.mongo_dbname);
+    app::MongoCollection collection = db[object_type];
+    timed_wait_for(
+        [&] {
+            uint64_t count = 0;
+            collection.count({}, [&](uint64_t c, util::Optional<app::AppError> error) {
+                REQUIRE(!error);
+                count = c;
+            });
+            if (count < expected_count) {
+                // querying the server too frequently makes it take longer to process the sync changesets we're
+                // waiting for
+                millisleep(100);
+                return false;
+            }
+            return true;
+        },
+        std::chrono::minutes(5));
```

Comment:
There's a lot of leeway to avoid spurious timeouts, but some of the audit tests do take over a minute since they're specifically testing the handling of large amounts of data.